### PR TITLE
[BE] Unify PG assignments

### PR DESCRIPTION
**Pull request by kwen2501**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #132230

python's `or` operator returns `bar` in cases of 
`foo = None or bar`

cc @XilunWu @H-Huang @awgu @wanchaol @fegin @fduwjj @wz337 @wconstab @d4l3k @c-p-i-o

### CODE REVIEW

**Review by Skylion007 — APPROVED** [review]

**Review by wconstab — APPROVED** [review]
💯

**Comment by kit1980** [comment]
I think it's more idiomatic for Python for explicitly check for None.
Otherwise there is an ambiguity if `group` is falsy, like if it's equal to 0 or to an empty string.

File: torch/distributed/distributed_c10d.py [hunk 1]
```diff
@@ -1190,10 +1190,7 @@ def get_backend_config(group: Optional[ProcessGroup] = None) -> str:
         The backend configuration of the given process group as a lower case string.
 
     """
-    if group is None:
-        pg = _get_default_group()
-    else:
-        pg = group
+    pg = group or _get_default_group()
```
Comment (kit1980) [inline torch/distributed/distributed_c10d.py:1193]: To retain the exact same semantic but still make it a bit shorter, can be `pg = _get_default_group() if group is None else group`

**Pull request closed without merging**